db/migrate/20230401123457_create_components.rb

```diff
@@ -0,0 +1,10 @@
+class CreateComponents < ActiveRecord::Migration[6.0]
+  def change
+    create_table :components do |t|
+      t.string :name, null: false
+      t.references :library, null: false, foreign_key: true
+
+      t.timestamps
+    end
```

Comment:
Consider adding an index to the `library_id` column to improve query performance, especially if the table grows large.